distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java

```diff
@@ -134,4 +137,32 @@ private static Stream<String> maybeWorkaroundG1Bug() {
         }
         return Stream.of();
     }
+
+    private static Stream<String> maybeAttachEntitlementAgent(boolean useEntitlements) {
+        if (useEntitlements == false) {
+            return Stream.empty();
+        }
+
+        Path dir = Path.of("lib", "entitlement-bridge");
+        if (Files.exists(dir) == false) {
+            throw new IllegalStateException("Directory for entitlement bridge jar does not exist: " + dir);
+        }
+        String bridgeJar;
+        try (var s = Files.list(dir)) {
+            var candidates = s.limit(2).toList();
```

Comment:
Why the `limit(2)` here?

Reply:
I only care if there are two or more. If there are ten thousand, I don't want to build the whole list first.

Reply:
I think that's fine but then the following exception will be a bit odd as it will always be `found 2` regardless of how many files actually exist.

Reply:
Also, FWIW, we're talking about code that runs exactly once, and on startup. Seems an unnecessary thing to optimize.